app/main.py

```
@@ -1,3 +1,62 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+
+        self.km = km
+
+    def __str__(self) -> str:
+
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        if other == 0:
+            print("error")
```

In the `__truediv__` method, division by zero is currently handled by printing an error message. It's better to raise an exception, such as `ZeroDivisionError`, to properly handle this error case and prevent undefined behavior.